### ISSUE TITLE
Restarting from previous run, order of atoms in potcar is incorrect

### ISSUE THREAD

**Issue by mdavezac** [pylada/pylada-light#15]
```
I noticed strange behavior when reading in results from a Relax() run to start a Vasp() run.
The order of the POTCAR rearranged, but the structure file stayed constant -- leading to atoms 
being in the wrong positions.  

I tried to identify the source of the error to no avail.  However, I was able to remedy the error by
using Relax() instead of Vasp() for the second run, then manually changing the settings to what
I needed (e.g. Relax().relaxation='static').
```

**Comment by vimtex** [comment]
Indeed.
It may be related to the usage of set in ./crystal/__init__.py.
The following seems a workaround, but I do not know how the original
specieset was used yet.

Another workaround may be to enforce calling vasp_ordered before calling
vasp() (Not tested yet).

def specieset(structure):
  """ Returns ordered set of species.

      Especially usefull with VASP since we are sure what the list of
species
      is always ordered the same way.
  """
 ## keep the original
order

  from collections import OrderedDict
  return list(OrderedDict.fromkeys([a.type for a in structure]))
 #return set([a.type for a in structure])


On Wed, Jun 15, 2016 at 7:08 AM, mdavezac <notifications@github.com> wrote:

> I noticed strange behavior when reading in results from a Relax() run to start a Vasp() run.
> The order of the POTCAR rearranged, but the structure file stayed constant -- leading to atoms
> being in the wrong positions.
>
> I tried to identify the source of the error to no avail.  However, I was able to remedy the error by
> using Relax() instead of Vasp() for the second run, then manually changing the settings to what
> I needed (e.g. Relax().relaxation='static').
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/pylada/pylada-light/issues/15>, or mute the thread
> <https://github.com/notifications/unsubscribe/AC4ZKx4v90IHZiqcLJk0wE1loDFdsYq1ks5qL90dgaJpZM4I2Pot>
> .
>



-- 
Haowei Peng

Department of Physics
Temple University

**Comment by mdavezac** [comment]
Actually, it looks like `vasp_ordered` and `specieset` are duplicate functionalities, but slightly different.
`specieset` is used by Vasp when creating the potcar and writing the poscar.
`vasp_ordered` is used nowhere that I can see.
@vimtex, @vstevano do you guys use `vasp_ordered` anywhere?
I might remove it otherwise.

**Comment by vimtex** [comment]
Hi Mayeul,

vasp_ordered are used in some htp.py driver script (at least the old ones).
Not a bad idea to keep it.
In a previous email, I mentioned use a list instead of set for the
specieset. Does that cause troubles?
Thanks.

Best, Haowei



On Sat, Jun 18, 2016 at 8:20 AM, mdavezac <notifications@github.com> wrote:

> Actually, it looks like vasp_ordered and specieset are duplicate
> functionalities, but slightly different.
> specieset is used by Vasp when creating the potcar and writing the poscar.
> vasp_ordered is used nowhere that I can see.
> @vimtex <https://github.com/vimtex>, @vstevano
> <https://github.com/vstevano> do you guys use vasp_ordered anywhere?
> I might remove it otherwise.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pylada/pylada-light/issues/15#issuecomment-226938596>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AC4ZKyfSHjX-5xWbUukAB7pUkZ8-J7dvks5qM-KhgaJpZM4I2Pot>
> .
>



-- 
Haowei Peng

Department of Physics
Temple University

**Comment by mdavezac** [comment]
resolved in df46c72
